refiners/array_axiom_enumerator.cpp

```diff
@@ -233,6 +233,7 @@ bool ArrayAxiomEnumerator::enumerate_axioms(const Term & abs_trace_formula,
 
     if (!found_lemmas) {
       // there appears to be a concrete counterexample
+      solver_->pop();
```

Comment:
we may want to store the counterexample. Where do we do that? If we pop now, we will not have access to the model values.

Reply:
That's a good point. I believe it's currently not handled at all (I don't think witnesses were handled before either). I'm not sure that's the right place anyway though since it's in the refiner. Maybe we should unroll the concrete transition system to get the trace when there's a counterexample? And if we want to save abstract model values to help guide it we could do that too, but there's currently no infrastructure for that. Btw, `witness` should probably be overridden in `CegProphecyArrays` so that it returns the right witness there.